test/exercises/duplicates/forStudents/README

```diff
@@ -0,0 +1,179 @@
+contents of the file expected-output. Additionally, you can check that your
+program is computing the correct SHA256 hashes for the files in that directory
+by running `sha256sum` and giving it the file to hash as an argument.
+
+
+Task 3: Compute Hashes in Parallel
+----------------------------------
+
+Adjust your implementation of `computeHashes` to compute file hashes
+in parallel. Does it still give correct output?
+
+To understand how this affects the performance of this program, add a timer to
+this function, using the Time module.
+
+  https://chapel-lang.org/docs/modules/standard/Time.html
+
+Then, test the performance of your parallel version by running it to
+find duplicate .chpl files in Chapel's test suite.
+
+  ./duplicates --filter=".chpl" <path-to-chapel-tests>
+
+Is it faster to compute the SHA256 sums in multiple tasks?
+Make sure to compile with --fast when doing performance comparisons!
+
+Task 4: Add Error Handling
+--------------------------
+
+We'd like our duplicate file detector to fail gracefully if it finds a
+file it cannot open. For example, if it runs accross a file it can't read,
+it will print an error like this:
+
+uncaught TaskErrors: 1 errors: PermissionError: Permission denied (in open with path "<some path>")
+  duplicates.chpl:30: thrown here
+  duplicates.chpl:30: uncaught here
+
+Instead of halting when such errors are encountered, make it instead
+log the errors. For the purposes of this exercise, let's assume that
+writing a message to stdout using `writeln` is sufficient.
+
+The first step is create a test directory in which you can see
+the above error. You can use the following commands to do so:
+
+  echo No Reading > example-files/noread
+  chmod a-r example-files/noread
+  # you can remove this file if you need to, with
+  #   rm example-files/noread
+
+Run your program on the example-files directory after adding this
+file. Your program should halt before printing any duplicates.
+
+Use error handling constructs to:
+ * output a message (we are imagining to a log file but for this exercise
+   just use writeln)
+ * continue processing the other files
+
+To learn more about error handling in Chapel, see the Error Handling
+primer:
+
+ https://chapel-lang.org/docs/primers/errorHandling.html
+
+
+Bonus Task B1: Create a Program Generating Each Byte
+----------------------------------------------------
+
+We used the file example-files/all-bytes1 in testing our file
+duplication detector. This file contains each byte once, and in order.
+To inspect this file, use xxd:
+
+  xxd example-files/all-bytes1
+
+This task is to create a Chapel program that can produce this file.
+You'll want to look at the I/O primer and module documentation:
+
+  https://chapel-lang.org/docs/primers/fileIO.html
+  https://chapel-lang.org/docs/modules/standard/IO.html
+
+Here are some hints of specific features you might use:
+ * `open`
+ * `file.writer` with `kind=iokind.native`
+ * the `%|u` format string
```

Comment:
I noticed our example did not use this format string - should we remove this hint?

Reply:
I could write one that works that way. I'd view it as a reasonable alternative. Seems like an OK hint.